### ISSUE TITLE
Redundant/ambiguous information in amici::Model

### ISSUE THREAD

**Issue by dweindl** [AMICI-dev/AMICI#2573]
Currently, `amici::Model::state_` and `amici::Model::simulation_parameters_` both contain `fixed_parameters` and `plist`. Those values aren't synced. `amici::Model::simulation_parameters_` is initialized during construction and never updated. Setters will only update `amici::Model::state_`.

This is confusing and likely to cause trouble at some point, and thus, should be changed.

**Comment by dweindl** [comment]
I think it might be best to always use `ExpData` or introduce a separate object for simulation settings (`t0`, `plist`, `pscale`, preequilibration/presimulation/simulation parameters, ...)  and completely get rid of `Model::simulation_parameters_`.
This would be a breaking change, but I think it would make things a bit cleaner. The whole preequilibration/presimulation/simulation is anyways managed outside `Model`, so `Model` doesn't need to know about any of that.
This would be a breaking change, though.

**Comment by FFroehlich** [comment]
Maybe we should plan towards making a 1.0.0 version with some other breaking changes

**Comment by dweindl** [comment]
Yes, I think that would make sense soon. Good to start collecting issues.
